app/models/shortener/shortened_url.rb

```diff
@@ -53,7 +53,10 @@ def _create_record
       super()
     rescue ActiveRecord::RecordNotUnique, ActiveRecord::StatementInvalid => err
       logger.info("Shortener gem attempted to generate a unique key and raised the following err: #{err}")
-      if (count +=1) < 10
+      ActiveRecord::Base.connection.execute 'ROLLBACK'
```

Comment:
1) I'd love if we could use https://api.rubyonrails.org/classes/ActiveRecord/Rollback.html instead of the raw sql `ROLLBACK`. In the docs I only see it used within an explicit `Model.transaction do` call, so this might not work 😕 . I haven't been able to get this to work locally outside of an explicit `.transaction do ... end` block, so we can probably ignore this idea
2) Should we make this conditional on whether an open transaction exists? Something like:
```suggestion
      if ActiveRecord::Base.connection.transaction_open?
        ActiveRecord::Base.connection.execute 'ROLLBACK'
      end
```

Reply:
I feel like we should try this and see what happens following the testing plan I laid out in the parent comment. Im happy to do this during my Monday support block if no one else picks it up before then

```suggestion
       if ActiveRecord::Base.connection.transaction_open?
         ActiveRecord::Rollback
       end
```